Cubical/HITs/GroupoidQuotients/Properties.agda

```diff
@@ -0,0 +1,133 @@
+{-
+
+Groupoid quotients:
+
+-}
+
+{-# OPTIONS --cubical --no-import-sorts --safe #-}
+module Cubical.HITs.GroupoidQuotients.Properties where
+
+open import Cubical.HITs.GroupoidQuotients.Base
+
+open import Cubical.Core.Everything
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.HLevels
+
+open import Cubical.Data.Sigma
+
+open import Cubical.Relation.Nullary
+open import Cubical.Relation.Binary.Base
+
+open import Cubical.HITs.PropositionalTruncation as PropTrunc using (∥_∥; ∣_∣; squash)
+open import Cubical.HITs.SetTruncation as SetTrunc using (∥_∥₂; ∣_∣₂; squash₂)
+
+-- Type quotients
+
+private
+  variable
+    ℓA ℓR ℓ : Level
+    A : Type ℓA
+    R : A → A → Type ℓR
+
+elimEq// : (Rt : BinaryRelation.isTrans R)
+          {B : A // Rt → Type ℓ}
+          (Bprop : (x : A // Rt) → isProp (B x))
+          {x y : A // Rt}
+          (eq : x ≡ y)
+          (bx : B x)
+          (by : B y) →
+          PathP (λ i → B (eq i)) bx by
+elimEq// Rt {B} Bprop {x = x} =
+  J (λ y eq → ∀ bx by → PathP (λ i → B (eq i)) bx by) (λ bx by → Bprop x bx by)
+
+elimProp : (Rt : BinaryRelation.isTrans R)
+         → {B : A // Rt → Type ℓ}
+         → ((x : A // Rt) → isProp (B x))
+         → ((a : A) → B [ a ])
+         → (x : A // Rt)
+         → B x
+elimProp Rt Bprop f [ x ] = f x
+elimProp Rt Bprop f (eq// {a} {b} r i) = elimEq// Rt Bprop (eq// r) (f a) (f b) i
+elimProp Rt Bprop f (comp// {a} {b} {c} r s i j) =
+  isSet→SquareP (λ i j → isProp→isSet (Bprop (comp// r s i j)))
+    (λ j → elimProp Rt Bprop f (eq// r j))
+    (λ j → elimProp Rt Bprop f (eq// (Rt a b c r s) j))
+    (λ i → f a)
+    (λ i → elimEq// Rt Bprop (eq// s) (f b) (f c) i)
+    i j
+elimProp Rt Bprop f (squash// x y p q r s i j k) =
+  isOfHLevel→isOfHLevelDep 3 (λ x → isSet→isGroupoid (isProp→isSet (Bprop x)))
+  _ _ _ _ (λ j k → g (r j k)) (λ j k → g (s j k)) (squash// x y p q r s) i j k
+  where
+    g = elimProp Rt Bprop f
+
+elimProp2 : (Rt : BinaryRelation.isTrans R)
+          → {C : A // Rt → A // Rt → Type ℓ}
+         → ((x y : A // Rt) → isProp (C x y))
+         → ((a b : A) → C [ a ] [ b ])
+         → (x y : A // Rt)
+         → C x y
+elimProp2 Rt Cprop f = elimProp Rt (λ x → isPropΠ (λ y → Cprop x y))
+                                   (λ x → elimProp Rt (λ y → Cprop [ x ] y) (f x))
+
+[]surjective : (Rt : BinaryRelation.isTrans R)
```

Comment:
State the type in terms of `isSurjection` from `Cubical.Functions.Surjection` and rename to `isSurjection[]`?

Reply:
Done